src/java.base/share/classes/java/lang/StackStreamFactory.java

```diff
@@ -81,11 +81,6 @@ private StackStreamFactory() {}
     @Native private static final int SHOW_HIDDEN_FRAMES        = 0x20;  // LambdaForms are hidden by the VM
     @Native private static final int FILL_LIVE_STACK_FRAMES    = 0x100;
 
-    /*
-     * For Throwable to use StackWalker, set useNewThrowable to true.
-     * Performance work and extensive testing is needed to replace the
-     * VM built-in backtrace filled in Throwable with the StackWalker.
-     */
     static final boolean isDebug =
```

Comment:
This property should probably be compared with true in a case insensitive manner. It may be better to use `Boolean::getBoolean` instead.

Justification:

JDK-8155775 (commit e8cd76568da1b32b26491e80f498cff1409336b7) removed the `getProperty` method which was previously used to read this property:

```java
private static boolean getProperty(String key, boolean value) {
        String s = GetPropertyAction.getProperty(key);
        if (s != null) {
            return Boolean.parseBoolean(s);
        }
        return value;
    }
```

`isDebug` was initialized with a false default if the system property was null:

```java
final static boolean isDebug = getProperty("stackwalk.debug", false);
``` 

I doubt the change in case handling was intentional, @cl4es may confirm.

Based on the above, I think we should consider using Boolean.getBoolean here, as that is case-insensitive and returns false for null:

```suggestion
    static final boolean isDebug = Boolean.getBoolean("stackwalk.debug");
```

Reply:
Changing the behavior is out of scope for this pr.